front/app/containers/Admin/reporting/components/ReportBuilder/TopBar/index.tsx

```diff
@@ -106,14 +111,36 @@ const ContentBuilderTopBar = ({
   };
 
   useEffect(() => {
-    if (draftEditorData !== undefined) {
+    let draftEditorDataLocale =
+      draftEditorData && selectedLocale
+        ? draftEditorData[selectedLocale]
+        : draftEditorData;
+    initialData = initialData === undefined ? {} : initialData;
```

Comment:
I really wouldn't reassign these variables- you're now changing their value for the whole component. It's generally bad practice to reassign arguments that you receive in a function. Instead I would do

I would instead remove the `hasChanges` as a state variable, and do

```ts
const hasChange = useMemo(() => {
  if (!selectedLocale || !draftEditorData) return false;
  const draftEditorDataLocale = draftEditorData[selectedLocale];

  return !isEqual(
    JSON.parse(JSON.stringify(initialData ?? {}),
    JSON.parse(JSON.stringify(draftEditorDataLocale ?? {})
  );
}, [selectedLocale, initialData, draftEditorData]);
```

Reply:
i need to be able to update `hasChange` on save. So keeping it as a state but taking your advice on reassigning variables.
```  useEffect(() => {
    if (!selectedLocale || !draftEditorData) return;
    setHasChange(
      isEqual(
        JSON.parse(JSON.stringify(initialData) ?? {}),
        JSON.parse(JSON.stringify(draftEditorData[selectedLocale] ?? {}))
      )
    )
  }, [initialData, draftEditorData]);```